src/test/java/teammates/test/cases/datatransfer/FeedbackQuestionAttributesTest.java

```diff
@@ -37,6 +40,230 @@ void setUpdatedAt(Instant updatedAt) {
 
     }
 
+    @Test
+    public void testValueOf() {
+        /* Solves the problem of "java.lang.NullPointerException: No API environment is registered for this thread on
+         Unit testing".
+         https://stackoverflow.com/questions/31994264/java-lang-nullpointerexception-no-api-environment-
+         is-registered-for-this-thread */
+        LocalServiceTestHelper helper = new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+        helper.setUp();
+
+        List<FeedbackParticipantType> participants = new ArrayList<>();
+        participants.add(FeedbackParticipantType.OWN_TEAM_MEMBERS);
+        participants.add(FeedbackParticipantType.RECEIVER);
+
+        List<FeedbackParticipantType> participantTypesAfterRemovingIrrelevantVisibilitiesOptions =
+                new ArrayList<>(participants);
+        participantTypesAfterRemovingIrrelevantVisibilitiesOptions.remove(FeedbackParticipantType.OWN_TEAM_MEMBERS);
+        participantTypesAfterRemovingIrrelevantVisibilitiesOptions.remove(FeedbackParticipantType.RECEIVER_TEAM_MEMBERS);
+
+        FeedbackQuestion qn = new FeedbackQuestion("test session", "cs1101", "test@example.com",
+                new Text("question text"), new Text("description"), 1, FeedbackQuestionType.TEXT,
+                FeedbackParticipantType.TEAMS, FeedbackParticipantType.TEAMS, 5, participants,
+                participants, participants);
+        qn.setId(12345678910L);
```

Comment:
We rely on `Objectify` Service to give a id to the object if the field attached with `@Id` is null. Therefore, there is no need to explicity test it (and change the production code).